hal/common/FileBase.cpp

```diff
@@ -18,24 +18,24 @@
 namespace mbed {
 
 FileBase *FileBase::_head = NULL;
-PlatformMutex FileBase::_mutex;
+SingletonPtr<PlatformMutex> FileBase::_mutex;
 
 FileBase::FileBase(const char *name, PathType t) : _next(NULL),
                                                    _name(name),
                                                    _path_type(t) {
-    _mutex.lock();
+    _mutex->lock();
```

Comment:
These mutex lock()/unlock() sequences **really** need to be encapsulated in a RAII class.